<a href="https://colab.research.google.com/github/DAGP1145/Mineriadedatos/blob/Control-de-cambios/Mineria_De_Datos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Fuente: https://exoplanetarchive.ipac.caltech.edu/index.html

Disponemos de muchas fuentes de informaciòn de las cuales teniamos dos principales opciones:

Sistema planetarios o sistemas planetarios compuestos.

La diferencia es que uno esta compuesto de un estudio en especifico por lo tanto lo datos son consistentes. Pero la desventaja es que podrian faltar datos porque simplemente ese estudio no midiò todo.

Por otro lado la segunda tabla esta compuesta de distintas publicaciones y las junta para crear una sola fila por planeta, por lo tanto es mas completa. El problema es que esa información puede no ser consistente, porque viene de distintos estudios que usaron métodos diferentes.

Se seleccionó la tabla de Parámetros Compuestos, ya que proporciona una única fila por planeta con un conjunto de datos más completo, lo que facilita el análisis y la aplicación de técnicas de minería de datos, aun cuando los datos no sean completamente consistentes entre sí.

In [ ]:
import pandas as pd # Estructura y manipulación de datos
import numpy as np #Programación vectorial

In [37]:
url = 'https://raw.githubusercontent.com/DAGP1145/Mineriadedatos/refs/heads/main/PSCompPars_2026.04.20_07.35.09.csv'
df = pd.read_csv(url, sep=",", comment="#")

# Exploración inicial de los datos.

Con info() Podremos visualizar las columnas, los tipos de datos, ademas de contabilizarlos.

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6160 entries, 0 to 6159
Data columns (total 91 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   pl_name          6160 non-null   object 
 1   hostname         6160 non-null   object 
 2   sy_snum          6160 non-null   int64  
 3   sy_pnum          6160 non-null   int64  
 4   sy_mnum          6160 non-null   int64  
 5   discoverymethod  6160 non-null   object 
 6   disc_year        6159 non-null   float64
 7   disc_facility    6160 non-null   object 
 8   disc_telescope   6160 non-null   object 
 9   disc_instrument  6160 non-null   object 
 10  pl_controv_flag  6160 non-null   int64  
 11  pl_orbper        5821 non-null   float64
 12  pl_orbpererr1    5331 non-null   float64
 13  pl_orbpererr2    5331 non-null   float64
 14  pl_orbperlim     5821 non-null   float64
 15  pl_orbsmax       5845 non-null   float64
 16  pl_orbsmaxerr1   3907 non-null   float64
 17  pl_orbsmaxerr2

Podriamos usar
df.columns para conocer las columnas. Tambien podriamos utilizar df.dtypes para conocer los tipos de datos de cada columa.

Pero con info es mas que suficiente para conocer ambas datos, además nos permite contabilizar los tipos de datos función que no hace con dtypes.

Una alternativa en tal caso de que no se este del todo conforme con lo mencionado puede ser: df.dtypes.value_counts()

In [28]:
df.shape()

(6160, 91)

df.shape para conocer las dimensiones de la base de datos. Manejamos 6160 filas con 91 columnas.

Necesitamos visualizar cuáles son las columnas en su conjunto que pertenecen a cierto tipo de dato.

In [43]:
df_objetos = df.select_dtypes(include=['object'])
df_int_floats = df.select_dtypes(include=['int', 'float'])

In [46]:
print(df_objetos.columns)
print('')
print(df_int_floats.columns)

Index(['pl_name', 'hostname', 'discoverymethod', 'disc_facility',
       'disc_telescope', 'disc_instrument', 'pl_bmassprov', 'st_spectype',
       'st_metratio', 'rastr', 'decstr'],
      dtype='object')

Index(['sy_snum', 'sy_pnum', 'sy_mnum', 'disc_year', 'pl_controv_flag',
       'pl_orbper', 'pl_orbpererr1', 'pl_orbpererr2', 'pl_orbperlim',
       'pl_orbsmax', 'pl_orbsmaxerr1', 'pl_orbsmaxerr2', 'pl_orbsmaxlim',
       'pl_rade', 'pl_radeerr1', 'pl_radeerr2', 'pl_radelim', 'pl_radj',
       'pl_radjerr1', 'pl_radjerr2', 'pl_radjlim', 'pl_bmasse',
       'pl_bmasseerr1', 'pl_bmasseerr2', 'pl_bmasselim', 'pl_bmassj',
       'pl_bmassjerr1', 'pl_bmassjerr2', 'pl_bmassjlim', 'pl_dens',
       'pl_denserr1', 'pl_denserr2', 'pl_denslim', 'pl_orbeccen',
       'pl_orbeccenerr1', 'pl_orbeccenerr2', 'pl_orbeccenlim', 'pl_insol',
       'pl_insolerr1', 'pl_insolerr2', 'pl_insollim', 'pl_eqt', 'pl_eqterr1',
       'pl_eqterr2', 'pl_eqtlim', 'ttv_flag', 'st_teff', 'st_tefferr1',
       'st_t